server/src/main/java/org/opensearch/indices/replication/common/ReplicationTarget.java

```diff
@@ -278,6 +281,12 @@ public abstract void writeFileChunk(
     );
 
     protected void closeInternal() {
-        store.decRef();
+        try {
+            store.decRef();
+        } finally {
+            if (indexShard.indexSettings().isRemoteStoreEnabled()) {
+                indexShard.remoteStore().decRef();
```

Comment:
Can this leave the store open and close the remote store in a non-happy case?

Reply:
Considering the dependency and linking of remote store - store - should it rather be 
```
 try {
            store.decRef();
            if (indexShard.indexSettings().isRemoteStoreEnabled()) {
                indexShard.remoteStore().decRef();
            }
       }
```

Reply:
if the store's refcount hits zero it will call closeInternal which throws if it can't close the directory:

```java
    private void closeInternal() {
        // Leverage try-with-resources to close the shard lock for us
        try (Closeable c = shardLock) {
            try {
                directory.innerClose(); // this closes the distributorDirectory as well
            } finally {
                onClose.accept(shardLock);
            }
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }
```
If that were to throw we'd still need to decref the remoteStore.